src/enc_effects.c

```diff
@@ -0,0 +1,366 @@
+    {HEARTHOME_GYM_ENC,     SEQ_BA_GYM},            // Fantina
+    {SNOWPOINT_GYM_ENC,     SEQ_BA_GYM},            // Candice
+    {CANALAVE_GYM_ENC,      SEQ_BA_GYM},            // Byron
+    {SUNYSHORE_GYM_ENC,     SEQ_BA_GYM},            // Volkner
+
+    // Elite four
+    {ELITE_FOUR_ENC_1,      SEQ_BA_TENNO},          // Aaron
+    {ELITE_FOUR_ENC_2,      SEQ_BA_TENNO},          // Bertha
+    {ELITE_FOUR_ENC_3,      SEQ_BA_TENNO},          // Flint
+    {ELITE_FOUR_ENC_4,      SEQ_BA_TENNO},          // Lucian
+
+    // Champion
+    {CHAMPION_ENC,          SEQ_BA_CHANP},          // Champion (Cynthia)
+
+    // Rival
+    {CATCHALL_ENC,          SEQ_BA_RIVAL},          // Rival (Barry/Cedric)
+    
+    // Legendary Pokemon
+    {LEGENDARY_POKE_ENC_1,  SEQ_BA_POKE},           // Shaymin
+    {LEGENDARY_POKE_ENC_2,  SEQ_BA_DPOKE2},         // Dialga, Palkia
+    {LEGENDARY_POKE_ENC_2,  SEQ_BA_DPOKE1},         // Uxie, Azelf
+    {CATCHALL_ENC,          SEQ_BA_DPOKE1},         // Mesprit (roaming)
+    {LEGENDARY_POKE_ENC_2,  SEQ_BA_SECRET1},        // Arceus
+    {LEGENDARY_POKE_ENC_1,  SEQ_BA_SECRET2},        // Regigigas, Heatran
+    {CATCHALL_ENC,          SEQ_BA_POKE},           // Cresselia (roaming)
+    {CATCHALL_ENC,          SEQ_BA_SECRET2},        // Articuno, Zapdos, Moltres (roaming)
+    {LEGENDARY_POKE_ENC_1,  SEQ_PL_BA_GIRA},        // Giratina
+    {LEGENDARY_POKE_ENC_1,  SEQ_PL_BA_REGI},        // Regirock, Regice, Registeel
+
+    // Team Galactic
+    {GALACTIC_GRUNT_ENC,    SEQ_BA_GINGA},          // Grunts
+    {GALACTIC_LEADER_ENC,   SEQ_BA_GINGA3},         // Commanders (Mars, Jupiter, Saturn)
+    {GALACTIC_LEADER_ENC,   SEQ_BA_AKAGI},          // Cyrus
+
+    // Other special encounter types
+    {FRONTIER_ENC,          SEQ_BA_TRAIN},          // Frontier trainers
+    {FRONTIER_ENC,          SEQ_BA_TRAIN},          // Link/Wi-Fi battles
+    {DOUBLE_ENC,            SEQ_BA_TRAIN},          // Double-battle against a trainer
+    {DOUBLE_ENC,            SEQ_BA_POKE},           // Wild encounters with a partner NPC
+    {FRONTIER_ENC,          SEQ_PL_BA_BRAIN},       // Frontier Brains (Palmer, Thorton, Argenta, Dahlia, Darach)
+    {DOUBLE_ENC,            SEQ_BA_GYM},            // Double-battle against a gym leader (Volkner+Flint, specifically)
+
+    // All other encounters
+    {CATCHALL_ENC,          SEQ_BA_TRAIN},          // Generic trainer
+    {CATCHALL_ENC,          SEQ_BA_POKE}            // Generic wild Pokemon
+};
+
+static u32 EncEffects_GetEffectsForTrainerClass(u32 trainerClass);
+static u32 EncEffects_GetEffectsForWildMon(Party * wildParty, int zoneID);
+static u32 EncEffects_GetEffectsForBattle(const UnkStruct_ov6_02240D5C * battleData);
+static u32 EncEffects_GetCutInFromEffects(u32 effectsID, const UnkStruct_ov6_02240D5C * battleData);
+static u32 EncEffects_GetBGMFromEffects(u32 effectsID, const UnkStruct_ov6_02240D5C * battleData);
```

Comment:
This is a personal opinion, not necessarily shared with the repo maintainers at this point, but I don't enjoy seeing "For", "From", "In" and such words in function names if it's not necessary. Also, seeing `Effects_GetXFromEffects` feels a bit redundant, though that could . As such, I'd propose something like:
```suggestion
static u32 EncEffects_GetTrainerClassEffect(u32 trainerClass);
static u32 EncEffects_GetWildMonEffect(Party * wildParty, int zoneID);
static u32 EncEffects_GetBattleEffect(const UnkStruct_ov6_02240D5C * battleData);
static u32 EncEffects_GetCutIn(u32 effectsID, const UnkStruct_ov6_02240D5C * battleData);
static u32 EncEffects_GetBGM(u32 effectsID, const UnkStruct_ov6_02240D5C * battleData);
```
I'd even suggest removing `Get` from the first three function names here.